compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -948,9 +944,12 @@ private void createLocalVariableTable(BIRFunction func, BIRVarToJVMIndexMap inde
         }
     }
 
-    private boolean isLocalOrArg(BIRVariableDcl localVar) {
+    private boolean isValidArg(BIRVariableDcl localVar) {
+        boolean localArg = localVar.kind == VarKind.LOCAL || localVar.kind == VarKind.ARG;
         boolean synArg = localVar.type.tag == TypeTags.BOOLEAN && localVar.name.value.startsWith("%syn");
-        return !synArg && (localVar.kind == VarKind.LOCAL || localVar.kind == VarKind.ARG);
+        boolean lambdaMapArg = localVar.metaVarName != null && localVar.metaVarName.startsWith("$map$block$") &&
```

Comment:
is there a consent in the frontend we can use for this?

Reply:
There's only a private constant in the frontend. And I didn't make it public